runtime/lua/vim/lsp/rpc.lua

```diff
@@ -50,9 +50,11 @@ recursive_convert_NIL = function(v, tbl_processed)
     return nil
   elseif not tbl_processed[v] and type(v) == 'table' then
     tbl_processed[v] = true
-    return vim.tbl_map(function(x)
-      return recursive_convert_NIL(x, tbl_processed)
-    end, v)
+    if not vim.tbl_islist(v) then
```

Comment:
This will also mean that dictionaries within tables will no longer be recursively processed. Maybe we should acount for that and only keep `vim.NIL` in lists, but still recurse into dictionary-style-tables ?

Reply:
Just pushed a changed which should allow trimming `vim.NIL` in dictionaries but spare lists 👍🏾